k2/csrc/fsa_equivalent.cc

```diff
@@ -123,4 +214,128 @@ bool RandomPath(const Fsa &a, Fsa *b,
   return true;
 }
 
+bool RandomPath(const Fsa &a, Fsa *b,
+                std::vector<int32_t> *state_map /*=nullptr*/) {
+  return RandomPath(a, b, false, state_map);
+}
+
+bool RandomPathWithoutEpsilonArc(
+    const Fsa &a, Fsa *b, std::vector<int32_t> *state_map /*= nullptr*/) {
+  return RandomPath(a, b, true, state_map);
+}
+
+bool Intersect(const Fsa &a, const float *a_weights, const Fsa &b, Fsa *c,
+               std::vector<float> *c_weights,
+               std::vector<int32_t> *arc_map_a /*= nullptr*/,
+               std::vector<int32_t> *arc_map_b /*= nullptr*/) {
+  CHECK_NOTNULL(c);
+  CHECK_NOTNULL(c_weights);
+  c->arc_indexes.clear();
+  c->arcs.clear();
+  c_weights->clear();
+  if (arc_map_a != nullptr) arc_map_a->clear();
+  if (arc_map_b != nullptr) arc_map_b->clear();
+
+  if (IsEmpty(a) || IsEmpty(b)) return true;
+  if (!IsArcSorted(a) || !IsArcSorted(b)) return false;
+  if (!IsEpsilonFree(b)) return false;
+
+  int32_t final_state_a = a.NumStates() - 1;
+  int32_t final_state_b = b.NumStates() - 1;
+  const auto arc_a_begin = a.arcs.begin();
+  const auto arc_b_begin = b.arcs.begin();
+  using ArcIterator = std::vector<Arc>::const_iterator;
+
+  const int32_t final_state_c = -1;  // just as a placeholder
```

Comment:
The name of a `const` variable has the style `kConst`

Reply:
Sure